plugins/module_utils/foreman_helper.py

```diff
@@ -270,6 +274,9 @@ def list_resource(self, resource, search=None, params=None):
             params = {}
         else:
             params = params.copy()
+
+        params = self.foremanapi.resource(resource).action('index').prepare_params(params)
```

Comment:
WIP because this cleaning should happen *after* we add search and per_page, but this breaks external_group and snapshot test which I didn't investigate yet